swiper.el

```diff
@@ -193,49 +199,85 @@ If the input is empty, select the previous history element instead."
+current Swiper input."
+  (interactive (list (when current-prefix-arg (caar query-replace-defaults))
+                     (when current-prefix-arg (cdar query-replace-defaults))
+                     (when (use-region-p) (region-beginning))
+                     (when (use-region-p) (region-end))))
+  (barf-if-buffer-read-only)
+  (if (window-minibuffer-p)
+      (ivy-exit-with-action
+       (lambda (x)
+         (swiper-isearch-action x)
+         (swiper-query-replace t)))
+    (when start
+      (deactivate-mark) ;; TODO: maybe add a fake active region as overlay
+      (goto-char start))
+    (when (string-or-null-p initial-regexp)
+      (let ((ivy-fixed-height-minibuffer t)
+            (cursor-in-non-selected-windows nil)
+            (swiper-min-highlight 1))
+        (ivy-read
+         "Swiper query replace: "
+         #'swiper-isearch-function
+         :initial-input initial-regexp
+         :keymap swiper-isearch-map
+         :dynamic-collection t
+         :require-match t
+         :action #'swiper-isearch-action
+         :re-builder #'swiper--regex-plain
+         :history query-replace-from-history-variable
+         :extra-props (list :fname (buffer-file-name))
+         :caller 'swiper-isearch)))
+    (when (string= "" ivy-text) (user-error "Empty input"))
+    (swiper--delayed-add-overlays)
+    (swiper--add-cursor-overlay (ivy-state-window ivy-last))
+    (swiper--query-replace-setup)
+    (unwind-protect
+        (let* ((from (ivy-re-to-str ivy-regex))
+               (ivy-count-format "")
+               (to
+                (query-replace-compile-replacement
+                 (ivy-read
+                  (format "Query replace %s with: " from) nil
+                  :initial-input initial-replacement
+                  :history query-replace-to-history-variable
+                  ;; TODO: M-n here could insert (map-elt query-replace-defaults from)
+                  :update-fn #'swiper--query-replace-updatefn)
+                 t)))
+          (swiper--cleanup)
+          (add-to-history 'query-replace-defaults (cons from to))
+          (with-ivy-window
+            (perform-replace from to t t nil nil nil
+                             (or start
+                                 (if (looking-back from (line-beginning-position))
+                                     (match-beginning 0)
+                                   (point)))
+                             (or end (point-max)))))
+      (swiper--cleanup)
+      (swiper--query-replace-cleanup))))
+
+(ivy-add-actions 'swiper '(("q" (lambda (x)
+                                  (swiper--action x)
+                                  (swiper-query-replace t))
+                            "query replace")))
+(ivy-add-actions 'swiper-isearch '(("q" (lambda (x)
+                                          (swiper-isearch-action x)
+                                          (swiper-query-replace t))
+                                    "query replace")))
```

Comment:
Can you please unquote the lambdas so that they're evaluated?

Reply:
Like so?

Reply:
Yep, that's what I meant. The alternative, if you prefer, is to use function symbols rather than anonymous functions, but either is fine.